lumicks/pylake/correlated_stack.py

```diff
@@ -17,10 +21,59 @@ class TiffFrame:
     def __init__(self, page):
         self._src = page
 
+    def align(self):
+        """ reconstruct image using alignment matrices from Bluelake 
+            return aligned image as a NumPy array 
+        """
+
+        def correct_alignment_offset(alignment, x_offset, y_offset):
+            # translate the origin of the image so that it matches that of the original transform
+            transform1 = np.eye(3)
+            transform1[0,-1] = -x_offset
+            transform1[1,-1] = -y_offset
+            # apply the original transform to the translated image. 
+            # it only needs to be resized from a 2x3 to a 3x3 matrix
+            transform2 = np.vstack((alignment, [0,0,1]))
+            # translate the image back to the original origin. 
+            # takes into account both the offset and the scaling performed by the first step
+            transform3 = np.eye(3)
+            transform3[0,-1] = transform2[0,0]*x_offset
+            transform3[1,-1] = transform2[1,1]*y_offset
+            # concatenate the transforms by multiplying their matrices and ignore the unused 3rd row
+            return np.dot(transform3, np.dot(transform2, transform1))[:2,:]
+
+
+        description = json.loads(self._src.description)
+        align_mats = [np.array(description[f"Alignment {color} channel"]).reshape((2,3))
+                      for color in ('red', 'blue')]
+        align_roi = np.array(description["Alignment region of interest (x, y, width, height)"])[:2]
+        roi = np.array(description["Region of interest (x, y, width, height)"])[:2]
+        
+        x_offset, y_offset = align_roi - roi
+        if not (x_offset == 0 and y_offset == 0):
+            align_mats = [correct_alignment_offset(mat, x_offset, y_offset) for mat in align_mats]
+
+        img = self.data
+        rows, cols, _ = img.shape
+
+        for mat, channel in zip(align_mats, (0, 2)):
+            img[:,:,channel] = cv2.warpAffine(img[:,:,channel], mat, (cols,rows),
+                                              flags=(cv2.INTER_LINEAR | cv2.WARP_INVERSE_MAP),
+                                              borderMode=cv2.BORDER_CONSTANT, borderValue=0)
+        return img
+        
     @property
     def data(self):
         return self._src.asarray()
 
+    @property
+    def is_rgb(self):
+        return self.data.ndim > 2
+
+    @property
+    def plot_data(self):
+        return (self.data/256).astype(np.uint8) if self.is_rgb else self.data
```

Comment:
I agree, the vmax doesn't do anything for rgb images since there's no colormap. But, as I noticed during testing, sometimes the images are quite dark if the full dynamic range isn't used. Perhaps including a special implementation for `vmax` in a `plot_rgb()` method that could renormalize the data to a specified range? I like the idea of including the single-channel plotting functions too